hudi-common/src/main/java/org/apache/hudi/common/model/HoodieRecordMerger.java

```diff
@@ -19,6 +19,10 @@
 package org.apache.hudi.common.model;
 
 import org.apache.avro.Schema;
```

Comment:
can we introduce a `HoodieSchema` class hierarchy. and make the API depend on that. Based on context, we would provide HoodieAvroSchema or HoodieSparkSchema etc?

Reply:
Do we need to complete schema unification in rfc46? The HoodieRecord API is used everywhere in Hudi code. If we want to use HoodieSchema, we need to almost remove avro schema in hudi code, it will be a big project.

Reply:
We've discussed this prior that it would be gigantic rename across the board, and agreed that it's probably better to be carried out separately